### PR TITLE
feat: revert last commit for lower cognitive load

### PR DESCRIPTION
**Pull request by adohe**
## What type of PR is this?
/kind feature

## What this PR does / why we need it:
revert last commit change for lower cognitive load
## Which issue(s) this PR fixes:

Fixes #

### Does this PR introduce a user-facing change?

```release-note

```

### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: models/schema/v1/workload/secret/secret.k [hunk 1]
```diff
@@ -37,197 +28,29 @@ schema BasicAuthSecret(SecretBase):
...
-    """
-
-    # The tls secret type.
-    type:                       "tls"
-
-    # Data contains certificate and its associated key.
-    data:                       {str:str}
-
-    check:
+        all k in data {
```
Comment (Peefy) [inline models/schema/v1/workload/secret/secret.k:47]: I would suggest using union types to implement different types of Secrets, which is very suitable for extension. Here is a code implementation.
```python
schema SecretBase:
    params?:  {str:str}
    immutable?:                 bool

schema Basic(SecretBase):
    username: str
    password: str

schema Token(SecretBase):
    token: str

schema Certificate(SecretBase):
    "tls.crt": str
    "tls.key": str

type Secret = Basic | Token | Certificate

schema Config:
    secret?: Secret
```

**Review by liu-hm19 — APPROVED** [review]
lgtm

File: models/schema/v1/workload/secret/secret.k (hunk 1 above)
Comment (Peefy) [reply models/schema/v1/workload/secret/secret.k:47]: Of course, if you still use the combination of type and data fields, I suggest defining a dictionary to maintain it.
```python
SECRET_TYPE_DATA_MAPPING = {
    basic = ["username", "password"]
    token = ["token"]
}

schema Secret:
    type: str
    data?: {str:str} = {}

    check:
          type in SECRET_TYPE_DATA_MAPPING, "type must be one of ${list(SECRET_TYPE_DATA_MAPPING)}"
          all k in data {
                k in SECRET_TYPE_DATA_MAPPING[type]
          }, "a valid secret data key name must be one of ${SECRET_TYPE_DATA_MAPPING[type]} for ${type} type secret, got ${list(data)}"
```

**Review by liu-hm19 — APPROVED** [review]
lgtm

**Pull request merged**